### PR TITLE
[doc] refactor python ndarray api

### PR DESCRIPTION
**Pull request by mli**
This PR adds more comments for ndarray operators and refactors ndarray API docs.

A preview is available at http://ec2-54-152-159-175.compute-1.amazonaws.com///api/python/ndarray.html

## refactor api/python/ndarray.md 

The refactor consists

1. operators that should be deprecated are removed from the summary list, including

```
true_divide
concatenate
identity
imdecode
argmax_channel
broadcast_*
choose_element_0index
fill_element_0index
onehot_encode
Crop
Sequence* // we should keep SequenceLast and SequenceMask, but I did not figure out how to use them.
```

2. we are agreed to use lower cases for array operators, so several alias are added.

```
ElementWiseSum  -> add_n
Reshape -> reshape
Flatten -> flatten
SliceChannel -> split
SwapAxis ->swapaxes
broadcast_axis -> broadcast_axes
Pad -> pad 
Concat -> concat 
```

3. bug fix for `rint` and `fix`. 

4. updated mxnet.css

## discussion to make the arguments be more consistent

For example, we used several input names including

```
lhs and rhs
src
a
data
```

we can use either 1) `data`, or `data0`, `data1` for multi-input, or 2) `a`, `b` to stay the same with numpy

Also we used `axis`, `axes`, `dim` to refer to dimensions. Suggest to use `axis` for single dimension and `axes` for multi-dimensions. 

some additional changes include `ret_typ` -> `ret_type` or `rtype` for `topk`.

## TODOs for next PRs

1. comments for neural network layers
2. improve `symbol.md`

### CODE REVIEW

File: src/operator/tensor/indexing_op.cc [hunk 1]
```diff
@@ -129,7 +186,7 @@ NNVM_REGISTER_OP(one_hot)
 .set_attr<nnvm::FInferType>("FInferType", OneHotOpType)
 .set_attr<FCompute>("FCompute<cpu>", OneHotOpForward<cpu>)
 .set_attr<nnvm::FGradient>("FGradient", MakeZeroGradNodes)
-.add_argument("indices", "NDArray", "array of locations where to set on_value")
+.add_argument("indices", "ndarray-or-symbol", "array of locations where to set on_value")
```
Comment (piiswrong) [inline src/operator/tensor/indexing_op.cc]: I think front end actually relies on this.

Comment (mli) [inline src/operator/tensor/indexing_op.cc]: you mean onehot needs ndarray indices? or this type info is required by frontend for most ops?

Comment (piiswrong) [inline src/operator/tensor/indexing_op.cc]: https://github.com/dmlc/mxnet/blob/master/python/mxnet/_ctypes/ndarray.py#L76

Changing it to NDArray-or-Symbol should make it backward compatible

**Comment by jermainewang** [comment]
Can we keep `concatenate`? Numpy has it https://docs.scipy.org/doc/numpy/reference/generated/numpy.concatenate.html. The same for `true_divide`, we could keep it for now although we didn't support integer array yet (we have `equal`, `less`, etc. that also only have floating number version).

**Comment by mli** [comment]
These functions still exist. I just removed them in the summary index.

**Comment by sergeykolychev** [comment]
@mli https://github.com/mli/mxnet/pull/2 this pull resolves perl build problem, use it please.

**Comment by mli** [comment]
since this PR does not change codes excepts documents, i'm going to merge it if there is no objection.

**Comment by piiswrong** [comment]
lint errors

**Comment by mli** [comment]
fixed lint errors caused by this PR. but there are other errors such as in `io.py`. i'm going to fix them in another PR.

**Pull request merged**